psi4/src/CMakeLists.txt

```diff
@@ -25,7 +25,11 @@ set(sources_list export_psio.cc
 add_library(core SHARED ${sources_list})
 
 get_property(psi4_binmodules GLOBAL PROPERTY BINLIST)
-target_link_libraries(core PRIVATE ${psi4_binmodules})
+if(ENABLE_psi4pasture)
+  target_link_libraries(core PRIVATE ${PRE_LIBRARY_OPTION} ${psi4_binmodules} ${POST_LIBRARY_OPTION})
```

Comment:
I'll have to devise a better way on this. Because as soon as PCMSolver is turned on, there'll be two sets of _LIBRARY_OPTIONs on the final link line, and I promise CMake won't leave them alone but will instead optimize them to one. :-( Do you know exactly which libs symbols are needed from?